src/step_definitions/elements.js

```diff
@@ -107,16 +107,16 @@ defineSupportCode(function ({ When, Then }) {
         if (!isPresent) {
```

Comment:
This is not how it should work. Right now it will continue to be executed even is element is no longer available. 

`sync` method should be called after clearInterval and should force getting out of the then callback (by calling `return;`)

If element is still visible, `maxRepeats` should be reduced by 1 and in case of it reaching 0 it should clear interval, call sync with error and also call `return`

Reply:
changes in new commit.

Reply:
@simonsipl look at @tgorskievo comment, you did almost what I wanted.